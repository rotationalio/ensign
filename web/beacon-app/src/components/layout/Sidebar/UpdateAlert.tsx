import { Trans } from '@lingui/macro';
import { Button } from '@rotational/beacon-core';

function UpdateAlert() {
  return (
    <>
      <p>
        <Trans>
          A new version of Ensign is available. Click <span className="font-bold">"Update"</span> to
          use the latest version.
        </Trans>
      </p>
      <div>
        <Button
          size="small"
          onClick={() => window.location.reload()}
          data-testid="update-alert-btn"
        >
          <Trans>Update</Trans>
        </Button>
      </div>
    </>
  );
}

export default UpdateAlert;
