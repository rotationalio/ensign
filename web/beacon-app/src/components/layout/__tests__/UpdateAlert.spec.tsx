// Test the UpdateAlert component
// Must mock the useFetchStatus hook
// The appConfig should be mocked as well
// If the version from the useFetchStatus hook is different from the version from the appConfig, the UpdateAlert should be rendered

import { render, screen } from '@testing-library/react';
import React from 'react';
import { describe, expect, it, vi } from 'vitest';

import UpdateAlert from '../Sidebar/UpdateAlert';

vi.mock('@/features/home/hooks/useFetchStatus', () => ({
  useFetchStatus: () => ({
    status: {
      version: '0.11.0',
    },
  }),
}));

vi.mock('@/appConfig', () => ({
  appConfig: {
    version: '0.10.0',
  },
}));

vi.mock('@lingui/macro', async (importOrginial: any) => ({
  ...importOrginial,
  t: () => vi.fn(),
  Trans: () => vi.fn(),
}));

const renderComponent = () => {
  return render(<UpdateAlert />);
};

describe('UpdateAlert', () => {
  it('should render if the version number returned from the useFetchStatus hook is different from the version number from the appConfig', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
    expect(screen.getByTestId('update-alert-btn')).toBeInTheDocument();
  });

  /* it('should refresh the page when the update button is clicked', () => {
        renderComponent();
        const updateBtn = screen.getByTestId('update-alert-btn');
        fireEvent.click(updateBtn);

        // Test reload functionality.
        expect(window.location.reload).toHaveBeenCalled();
    }); */
});
