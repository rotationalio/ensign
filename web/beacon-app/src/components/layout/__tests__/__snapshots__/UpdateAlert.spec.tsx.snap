// Vitest Snapshot v1

exports[`UpdateAlert > should render if the version number returned from the useFetchStatus hook is different from the version number from the appConfig 1`] = `
<div>
  <p />
  <div>
    <button
      class="sc-bcXHqe llTXds line-height-1.75 cursor-pointer text-white font-size-14 px-4 py-2 min-h-[28px] bg-inherit transition-colors duration-200 ease-in-out rounded-5 focus:outline-none text-[14px] font-bold"
      data-testid="update-alert-btn"
    />
  </div>
</div>
`;
